ddtrace/_trace/span.py

```diff
@@ -634,6 +650,24 @@ def set_link(
             attributes=attributes,
         )
 
+    def set_pointer(
+        self,
+        kind: str,
+        direction: SpanPointerDirection,
+        hash: SpanPointerHash,
+        **extra_attributes: Any,
+    ) -> None:
+        pointers_key = PointersKey(hash, direction)
+
+        if pointers_key in self._pointers:
+            log.debug(
+                "Span %d already has a pointer %s. Overwriting existing pointer: %s",
+                self.span_id,
+                str(pointers_key),
+                str(self._pointers[pointers_key]),
+            )
+        self._pointers[pointers_key] = SpanPointer(kind=kind, direction=direction, hash=hash, extra_attributes=extra_attributes)
```

Comment:
since `SpanPointer`s are serialized to `span_links`, it might be nice to just store them in the `_links` property instead. But that runs up against the checks we have on `SpanLink`s (the trace and span ids must not be `0` or `None`, and we can only have one link per `span_id`). So we store them separately.